From 39f519ca216a525b358b10ab826b0c4e0d46f145 Mon Sep 17 00:00:00 2001
From: "Nishioka, Toshiki" <toshiki.nishioka@intel.com>
Date: Thu, 10 Jun 2021 11:19:13 +0900
Subject: [PATCH 106/122] igc: Fix null pointer dereference upon driver probe

Avoid using IGC_MAX_TX_QUEUES when iterating tx_ring queue upon driver
probe otherwise it hits kernel null pointer dereference bug when available
on-line cpu count is less than 4.

Signed-off-by: Nishioka, Toshiki <toshiki.nishioka@intel.com>
Tested-by: Muhammad Husaini Zulkifli <muhammad.husaini.zulkifli@intel.com>
Signed-off-by: Muhammad Husaini Zulkifli <muhammad.husaini.zulkifli@intel.com>
---
 drivers/net/ethernet/intel/igc/igc_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/intel/igc/igc_main.c b/drivers/net/ethernet/intel/igc/igc_main.c
index 755b5623173a..1dcda6bcb401 100644
--- a/drivers/net/ethernet/intel/igc/igc_main.c
+++ b/drivers/net/ethernet/intel/igc/igc_main.c
@@ -5542,7 +5542,7 @@ static int igc_tsn_clear_schedule(struct igc_adapter *adapter)
 	adapter->base_time = 0;
 	adapter->cycle_time = NSEC_PER_MSEC;
 
-	for (i = 0; i < IGC_MAX_TX_QUEUES; i++) {
+	for (i = 0; i < adapter->num_tx_queues; i++) {
 		struct igc_ring *ring = adapter->tx_ring[i];
 
 		ring->start_time = 0;
-- 
2.17.1

